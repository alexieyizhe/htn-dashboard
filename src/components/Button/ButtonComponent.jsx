import React from "react";
import styled from "styled-components";


const Container = styled.button`
  cursor: pointer;
  padding: 1em 2em;
  border-radius: ${props => props.theme.defaults.borderRadiusRound};

  color: white;
  background-color: ${props => props.backgroundColor || props.theme.colors.lightBlack};

  &:focus {
    outline: none;
  }

  transition: background-color 250ms ease-in-out;
  &:hover {
    background-color: ${props => props.backgroundColor || props.theme.colors.black};
  }
`;


const Button = ({
  label,
  backgroundColor,
  type = "button",
  onClickHandler = null,
}) => (
  <Container
    type={type}
    backgroundColor={backgroundColor}
    onClick={onClickHandler !== null ? () => onClickHandler() : () => console.log(`clicked ${label} button`)}
  >
    {label}
  </Container>
);


export default Button;
