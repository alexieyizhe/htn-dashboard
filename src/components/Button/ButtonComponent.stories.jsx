import React from "react";
import { storiesOf } from "@storybook/react";
import Button from "./ButtonComponent";


storiesOf("Button", module)
  .add("Default", () => <Button label="Default" />)
  .add("Custom Color", () => <Button backgroundColor="rgb(146, 160, 241)" label="Blue!" />);
