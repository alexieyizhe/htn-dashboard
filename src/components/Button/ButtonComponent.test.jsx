/* eslint-disable no-undef */
import React from "react";

import Button from "./ButtonComponent";

describe("Button", () => {
  it("renders correctly", () => {
    const buttonComponent = mountWithTheme(<Button label="Default" />);
    expect(buttonComponent).toMatchSnapshot();
  })
})
