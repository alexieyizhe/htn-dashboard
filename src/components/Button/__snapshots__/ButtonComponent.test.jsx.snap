// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  cursor: pointer;
  padding: 1em 2em;
  border-radius: 50px;
  color: white;
  background-color: #696969;
  -webkit-transition: background-color 250ms ease-in-out;
  transition: background-color 250ms ease-in-out;
}

.c0:focus {
  outline: none;
}

.c0:hover {
  background-color: #4B4B4B;
}

<Button
  label="Default"
>
  <styled.button
    onClick={[Function]}
    type="button"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  cursor: pointer;
  padding: 1em 2em;
  border-radius: ",
              [Function],
              ";

  color: white;
  background-color: ",
              [Function],
              ";

  &:focus {
    outline: none;
  }

  transition: background-color 250ms ease-in-out;
  &:hover {
    background-color: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.button",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "button",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      onClick={[Function]}
      type="button"
    >
      <button
        className="c0"
        onClick={[Function]}
        type="button"
      >
        Default
      </button>
    </StyledComponent>
  </styled.button>
</Button>
`;
